### PR TITLE
feat: added ibc responses

### PR DESCRIPTION
**Pull request by swelf19**
Changes in the main wasmd repo requires to update our contracts.
We can not parse events in the reply handler anymore, since there are no ones.
Now we have to get useful(sequence_id) payload from the reply msg data.
linked PR: https://github.com/neutron-org/neutron/pull/47

### CODE REVIEW

File: contracts/neutron_interchain_txs/src/contract.rs [hunk 1]
```diff
@@ -278,6 +277,8 @@ fn execute_clean_ack_results(deps: DepsMut) -> StdResult<Response<NeutronMsg>> {
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn sudo(deps: DepsMut, env: Env, msg: SudoMsg) -> StdResult<Response> {
+    deps.api
+        .debug(format!("WASMDEBUG: sudo: received sud msg: {:?}", msg).as_str());
```
Comment (oopcode) [inline contracts/neutron_interchain_txs/src/contract.rs:281]: sud

File: contracts/neutron_interchain_txs/src/contract.rs [hunk 2]
```diff
@@ -429,7 +430,19 @@ fn sudo_error(deps: DepsMut, request: RequestPacket, details: String) -> StdResu
 
 fn prepare_sudo_payload(mut deps: DepsMut, _env: Env, msg: Reply) -> StdResult<Response> {
     let payload = read_reply_payload(deps.storage)?;
-    let (channel_id, seq_id) = parse_sequence(deps.as_ref(), msg)?;
+    let resp: MsgSubmitTxResponse = serde_json_wasm::from_slice(
```
Comment (oopcode) [inline contracts/neutron_interchain_txs/src/contract.rs:433]: why did you decide to get rid of the parse_sequence() fn here?

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

File: contracts/ibc_transfer/src/contract.rs [hunk 3]
```diff
@@ -82,7 +85,17 @@ fn msg_with_sudo_callback<C: Into<CosmosMsg>>(
...
-    let (channel_id, seq_id) = parse_sequence(deps.as_ref(), msg)?;
+    let resp: MsgTransferResponse = ProtoMessage::parse_from_bytes(
+        msg.result
+            .into_result()
+            .map_err(StdError::generic_err)?
+            .data
+            .unwrap()
+            .as_slice(),
+    )
+    .unwrap();
```
[the remark this answers is not in the document]
Comment (foxpy) [reply contracts/ibc_transfer/src/contract.rs:96]: Where is our unwrap-forbidding lint?

[the remark this answers is not in the document]
Comment (foxpy) [reply contracts/ibc_transfer/src/contract.rs:96]: Please add `#![warn(clippy::unwrap_used, clippy::expect_used)]` to `contracts/ibc_transfer/src/lib.rs`

File: contracts/neutron_interchain_txs/src/contract.rs [hunk 4]
```diff
@@ -429,7 +430,19 @@ fn sudo_error(deps: DepsMut, request: RequestPacket, details: String) -> StdResu
...
+        msg.result
+            .into_result()
+            .map_err(StdError::generic_err)?
+            .data
+            .ok_or_else(|| StdError::generic_err("no result"))?
+            .as_slice(),
+    )
+    .map_err(|e| StdError::generic_err(format!("failed to parse response: {:?}", e)))?;
+    deps.api
+        .debug(format!("WASMDEBUG: reply msg: {:?}", resp).as_str());
```
Comment (foxpy) [inline contracts/neutron_interchain_txs/src/contract.rs:443]:
```suggestion
        .debug(&format!("WASMDEBUG: reply msg: {:?}", resp));
```

File: build_proto.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+docker build thirdparty/protoc-image  -t protoc_builder
+
+docker run --rm \
+-v $PWD:/opt \
+protoc_builder sh -c "protoc \
+-I/opt/proto \
+/opt/proto/transfer/v1/*.proto \
+--rust_out /opt/packages/neutron-sdk/src/ibc/"
```
Comment (foxpy) [inline build_proto.sh:8]: Please add `make proto-gen` rule to run this script

**Review by foxpy — CHANGES_REQUESTED** [review]

**Review by foxpy — APPROVED** [review]

**Review by oopcode — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**